app/src/main/java/fr/free/nrw/commons/nearby/NearbyMapFragment.java

```diff
@@ -385,10 +390,65 @@ private void passInfoToSheet(Place place) {
         commonsButton.setEnabled(place.hasCommonsLink());
         commonsButton.setOnClickListener(view -> openWebView(place.siteLinks.getCommonsLink()));
 
-        icon.setImageResource(place.getDescription().getIcon());
-        description.setText(place.getDescription().getText());
-        title.setText(place.name);
-        distance.setText(place.distance);
+        icon.setImageResource(place.getLabel().getIcon());
+
+        description.setText(place.getLongDescription());
+        title.setText(place.name.toString());
+        distance.setText(place.distance.toString());
+
+        fabCamera.setOnClickListener(view -> {
+            Timber.d("Camera button tapped. Image title: " + place.getName() + "Image desc: " + place.getLongDescription());
+            controller = new ContributionController(this);
+            DirectUpload directUpload = new DirectUpload(place.getName(), place.getLongDescription(), this, controller);
```

Comment:
Pass in your dependencies, or, manually setup Dagger injection here.

Reply:
Done, thanks!